pkg/standalone/standalone.go

```diff
@@ -554,34 +574,36 @@ func installPlacement(wg *sync.WaitGroup, errorChan chan<- error, info initInfo)
 		return
 	}
 
-	err := installBinary(info.runtimeVersion, placementServiceFilePrefix, cli_ver.DaprGitHubRepo, info.fromDir)
+	err := installBinary(info.runtimeVersion, placementServiceFilePrefix, cli_ver.DaprGitHubRepo, info)
 	if err != nil {
 		errorChan <- err
 	}
 }
 
-func installBinary(version, binaryFilePrefix string, githubRepo string, fromDir string) error {
+// installBinary installs the daprd, placement or dashboard binaries and associated files inside the default dapr bin directory.
+func installBinary(version, binaryFilePrefix, githubRepo string, info initInfo) error {
 	var (
 		err      error
 		filepath string
 	)
 
 	dir := defaultDaprBinPath()
-	if fromDir == "" {
+	if isAirGapInit(info.fromDir) {
+		filepath = path_filepath.Join(info.fromDir, daprBinarySubDir, binaryName(binaryFilePrefix))
+	} else {
 		filepath, err = downloadBinary(dir, version, binaryFilePrefix, githubRepo)
 		if err != nil {
 			return fmt.Errorf("error downloading %s binary: %w", binaryFilePrefix, err)
 		}
-	} else {
-		filepath = path_filepath.Join(fromDir, daprBinarySubDir, binaryName(binaryFilePrefix))
 	}
 
 	extractedFilePath, err := extractFile(filepath, dir, binaryFilePrefix)
 	if err != nil {
 		return err
 	}
 
-	if fromDir == "" {
+	// remove downloaded archive from the default dapr bin path.
+	if !isAirGapInit(info.fromDir) {
```

Comment:
can this be moved to defer, so it doesn't slow down the init

Reply:
please create a separate issue for this change along with the comment below.

Reply:
taken care